cli/internal/utils/auth.go

```diff
@@ -0,0 +1,36 @@
+package utils
+
+import (
+	"github.com/docker/cli/cli/config"
+	"github.com/docker/cli/cli/config/types"
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/name"
+	"log"
+	"os"
+)
+// Login adds the given creds to the user's Docker config, usually located at $HOME/.docker/config.yaml. It does not try
+// to connect to the given registry, it just simply adds another entry to the config file.
+// This function was mostly adapted from https://github.com/google/go-containerregistry/blob/main/cmd/crane/cmd/auth.go
```

Comment:
If we link to an external repo it might be helpful to put a release/commit SHA. That project seems to get updated regularly and might be useful in the future to know exactly what the inspiration was.